std/traits.d

```diff
@@ -9261,6 +9265,7 @@ package(std) template DeducedParameterType(T)
     }
 
     static assert(is(DeducedParameterType!NoCopy == NoCopy));
+    static assert(is(DeducedParameterType!(inout(NoCopy)) == inout(NoCopy)));
```

Comment:
This test case passes already on Phobos `master`. What's the purpose of including it in this PR?

Reply:
To ensure `DeducedParameterType(alias T)` is not added directly to `DeducedParameterTypeImpl`, which makes this false, a unwanted behavior.